### DISCUSSION TITLE
onSend handler is never called

### DISCUSSION THREAD

**Discussion by ptomask** [statelyai/xstate#1881] [Q&A]
I wanted to register handler for the machine sending events to react upon them externally using this method of `Interpreter`: https://xstate.js.org/api/classes/interpreter.html#onsend

I expect it to trigger the handler whenever the machine being interpreted issues some `send` action as described here: https://xstate.js.org/docs/guides/actions.html#send-action

However the registered handler never gets called. Brief look over the `Interpreter`'s source seems like `sendListeners` `Set` is not read anywhere.

Am I missing something? If not, shall I try to make a PR?

**Comment by Andarist** [comment]
Wouldn't this suffice your needs?
```js
service.onTransition((_, event) => /* do smth with the event */)
```
You could also try to use `onEvent` right now.

@davidkpiano do we want to keep those extra `on*` methods? Can't most of them be handled through `onTransition`/`subscribe`?

**Reply by davidkpiano** [reply]
> @davidkpiano do we want to keep those extra `on*` methods?

No, and they're already deleted in the v5 `next` branch 👍 

> Can't most of them be handled through `onTransition`/`subscribe`?

Yes, absolutely (just like you pointed out) 👍
